### PR TITLE
support BB Q10 keyboard

### PR DESCRIPTION
**Pull request by mverch67**
LILYGO BB Q10 keyboard support, see #2311

Arrow key simulation:
```
UP: SYM SHIFT+2
DOWN: SYM SHIFT+8
LEFT: SYM SHIFT+4
RIGHT: SYM SHIFT+6
```
(first press SYM and release it, then SHIFT plus the number key at the same time)

### CODE REVIEW

**Comment by mverch67** [comment]
@rcarteraz I created this PR, please install the firmware from the .zip and check if it works for you!

**Review by caveman99 — APPROVED** [review]
LGTM :-) But see comment.

**Comment by caveman99** [comment]
Still lookin good 👍

**Comment by mverch67** [comment]
> @mverch67 Detected but not working.


Did you enabled the CannedMessageModule???

**Comment by mverch67** [comment]
> Yes to both

Do you see anything in the debug log when typing on the keyboard?

**Comment by mverch67** [comment]
> Checked for that too, and nothing that indicates a button has been pressed. Just the typical stuff.

Maybe the BBQ10 requires an additional I2C driver, e.g. https://github.com/arturo182/arduino_bbq10kbd
It could be that something like this is implemented in the esp32-C3 of the T-Deck...

**Comment by caveman99** [comment]
Looks like the older BBQ10 Lilygo Firmware is based on this code ... https://github.com/solderparty/bbq10kbd_i2c_sw - it's queried differently than the T-Deck firmware or the cardKB. Best bet would really to include the access library from the T-Watch firmware and modify the kb driver to use that. if 1F is detected.

**Comment by caveman99** [comment]
@rcarteraz that is to be expected, it's a rough approximation of what 'should' work. It seems with the older code the firmware needs to do the debouncing. I have found sample code for these and will try to improve them ... in the meantime can you check the keymapping is correct? e.g. no matter how often, but it should be the right key.

File: src/input/kbI2cBase.cpp [hunk 1]
```diff
@@ -66,47 +66,50 @@ int32_t KbI2cBase::runOnce()
...
-                break;
-            case 0x16: // Right (sym shift+6)
-                e.inputEvent = meshtastic_ModuleConfig_CannedMessageConfig_InputEventChar_RIGHT;
-                e.kbchar = 0x00; // tweak for destSelect
-                break;
-            case 0x0d: // Enter
-                e.inputEvent = meshtastic_ModuleConfig_CannedMessageConfig_InputEventChar_SELECT;
-                break;
-            case 0x00: // nopress
+            if ((key.key != 0x00) && (key.state == BBQ10Keyboard::StateRelease)) {
```
Comment (mverch67) [inline src/input/kbI2cBase.cpp:69]: I think this will never be true :)

**Comment by mverch67** [comment]
I was thinking about something like this:

```
BBQ10Keyboard::KeyEvent BBQ10Keyboard::keyEvent() const
{
    KeyEvent event = {.key = '\0', .state = StateIdle};

    if (keyCount() == 0)
        return event;

    const uint16_t buf = readRegister16(_REG_FIF);
    char key = buf >> 8;
    KeyState state = KeyState(buf & 0xFF);

    if (key == KEY_MOD_SYM && state == StateRelease) {
        sym_pressed = true;
        return event;
    }

    if (sym_pressed && key > 0x20) {
        sym_pressed = false;
        event.key = key - 0x20;
    }
    else {
        event.key = key;
    }
    event.state = state;

    return event;
}
```

**Pull request merged**